kernels/simple_matmul/main.c

```diff
@@ -149,7 +99,7 @@ int main() {
 
   int nerr = 0;
   for (int i = 0; i < M_size * N_size; i++) {
-    int32_t error = memrefC.aligned_data[i] - C_golden[i];
```

Comment:
Is this change necessary?

Reply:
yes! memrefC still has the weird output layout of the accelerator. The DMA transfer then also transforms the data when copying it to C, where it has the same layout as C_golden

Reply:
And all transforms should have append by the end of leaving function 
`_mlir_ciface_simple_matmul(&memrefA, &memrefB, &memrefC);` no?